src/App.tsx

```diff
@@ -1,14 +1,58 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React,
+{
+  useState,
+  useEffect,
+  useMemo,
+} from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { FilterBy } from './types/FilterBy';
+import { getTodos } from './api';
+import { getVisibleTodos } from './helpers';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [isError, setIsError] = useState(false);
+  const [selectedTodoId, setSelectedTodoId] = useState(0);
+  const [query, setQuery] = useState('');
+  const [filterBy, setFilterBy] = useState(FilterBy.All);
+
+  const getTodosFromApi = async () => {
+    try {
+      const todosFromApi = await getTodos();
+
+      setTodos(todosFromApi);
+    } catch {
+      setIsError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    getTodosFromApi();
+  }, []);
+
+  const visibleTodos = getVisibleTodos(todos, query, filterBy);
+
+  const changeSelectedTodo = (id: number) => {
+    setSelectedTodoId(id);
+  };
+
+  const activeTodo = useMemo(() => (
+    todos.find((todo) => todo.id === selectedTodoId)
+  ), [selectedTodoId]);
```

Comment:
Check my prev comment
![image](https://github.com/mate-academy/react_dynamic-list-of-todos/assets/84296041/f182daf7-fb77-45af-98ae-afce786b1033)